homeassistant/components/matter/sensor.py

```diff
@@ -448,4 +448,18 @@ def _update_from_device(self) -> None:
         entity_class=MatterSensor,
         required_attributes=(NeoCluster.Attributes.Current,),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.SENSOR,
+        entity_description=MatterSensorEntityDescription(
+            key="SwitchCurrentPosition",
```

Comment:
We didn't add tests for this new sensor although the code is probably exercised by the event tests. We should add a specific sensor test with some assertion.

Reply:
there are existing tests that cover this, no need to add a test for every discovery schema if the heuristics match existing sensor types

Reply:
I think we should have tests for every schema. Otherwise we could remove this schema without a test failure.

Reply:
I don't agree as we will have schemas for sensors for attributes with the same behavior.
So sensors that are very much alike but just discovered from another attribute (but have exactly the same behavior already covered in a test) should not need a special test in all cases because;

a) we will have 10000 tests
b) we simply do not have fixtures for everything as we're just working our way through the specifications and add theoretical discovery schemas for future devices.

I do agree that we NEED at least one test per "type" of schema/entity and prefer to have tests for special cases and/or when we have a fixture available.

Reply:
I think you're exaggerating that number and it's not a problem to have many tests.

If we can add theoretical schemas we can add theoretical fixtures.

Reply:
Sounds good! 👍